src/v/storage/index_state.cc

```diff
@@ -726,11 +726,9 @@ index_state::find_below_size_bytes(size_t distance) {
         return std::nullopt;
     }
     auto it = index.position_upper_bound(distance);
-    if (it == std::nullopt) {
-        return std::nullopt;
-    }
 
-    auto ix = it.value();
+    // Invariant: index.size() > 0
+    auto ix = it.value_or(index.size() - 1);
     if (ix > 0) {
         ix--;
```

Comment:
are you double decrementing if the `it` is nullopt?

its unfortunate that the helper routines are returning an optional type. they have a sort of doubling effect on the size of the state space we have to think about.

Reply:
fixed